### DISCUSSION TITLE
Reading audio data from aaf file

### DISCUSSION THREAD

**Discussion by ginwakeup** [AcademySoftwareFoundation/OpenTimelineIO#909] [Q&A] [answered]
Hello,
I've exported a protools session to an .aaf file, embedding all the audio data in it.

I am now trying to use opentimelineio to read this audio data from the aaf, but I can't find a way of doing this (if there is).
Any suggestions?

Thanks.

**Comment by jminor** [comment, marked as answer]
Hi @ginwakeup, OpenTimelineIO doesn't handle embedded media to or from AAF. We might add that in the future but at the moment it only handles the composition, not the media.

The underlying python module ( https://github.com/markreidvfx/pyaaf2 ) that OTIO uses for working with AAF does support embedded media, so this should be possible to add. One other company we know of has been looking into this recently, but I don't know of the status of their investigation.

If you're willing/able to look into adding that feature, we would be happy to assist with some guidance. The feature would need to be added as an optional argument in the AAF adapter here: https://github.com/PixarAnimationStudios/OpenTimelineIO/blob/master/contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py

**Reply by ginwakeup** [reply]
Thank you very much for your reply, that's very bad to know :( I'll try to sort something through pyaaf.
